aiida/manage/manager.py

```diff
@@ -93,127 +85,49 @@ def get_profile() -> Optional['Profile']:
         from .configuration import get_profile
         return get_profile()
 
-    def unload_backend(self) -> None:
-        """Unload the current backend and its corresponding database environment."""
-        manager = self.get_backend_manager()
-        manager.reset_backend_environment()
-        self._backend = None
-
-    def _load_backend(self, schema_check: bool = True, repository_check: bool = True) -> 'Backend':
-        """Load the backend for the currently configured profile and return it.
-
-        .. note:: this will reconstruct the `Backend` instance in `self._backend` so the preferred method to load the
-            backend is to call `get_backend` which will create it only when not yet instantiated.
+    @property
+    def backend_loaded(self) -> bool:
+        """Return whether a storage backend has been loaded.
 
-        :param schema_check: force a database schema check if the database environment has not yet been loaded.
-        :param repository_check: force a check that the database is associated with the repository that is configured
-            for the current profile.
-        :return: the database backend.
+        :return: boolean, True if database backend is currently loaded, False otherwise
         """
-        from aiida.backends import BACKEND_DJANGO, BACKEND_SQLA, get_backend_manager
-        from aiida.common import ConfigurationError, InvalidOperation
+        from aiida.manage import configuration
+        return configuration.STORAGE_BACKEND is not None
+
+    def get_backend(self) -> 'Backend':
```

Comment:
Where did the functionality go with loading a backend without validating the schema? Did it move elsewhere or did it get completely removed. Surely there are still places where it is necessary to load the backend but without validating the schema. Or does the validation now have to be called manually?

Reply:
> Surely there are still places where it is necessary to load the backend but without validating the schema.

Nope none; you only ever initialise the backend with a "valid" profile, i.e. you can connect to the storage and its at the head version (and specific to the `PsqlDostore`, the container.id matches the `repository|uuid` in the database). The backend instance would essentially be useless without this.

> Or does the validation now have to be called manually

Validation is performed simply by trying to initialise the backend `Backend(profile)`, if it is not valid it will error

Any functionality required to be run on a potentially invalid profile are available as `Backend` class methods, e.g. `Backend.migrate(profile)`

Reply:
Note, we could "lazy" validate, but even if we really wanted to, it would require #5172, i.e. when the backend has "full control" over access to the storage, and there are no parts of the code by-passing it, by accessing `get_scoped_session`

Reply:
Well I think you have already found an example where you need to work around this limitation (you added the explicit `options.PROFILE` to the devel sql run command). Will try to make inventory of the places where we used loading the backend with validation and see if they are still all covered.

Reply:
> Well I think you have already found an example where you need to work around this limitation (you added the explicit options.PROFILE to the devel sql run command)

Its not a workaround, thats literally the point of that command; to run SQL on the database, outside of any aiida logic. Its what I added the command to do (only a few months ago).

> Will try to make inventory of the places where we used loading the backend with validation and see if they are still all covered.

I'm still not sure if you are understanding this, here is the code of `PsqlDostoreBackend.__init__`:

```python
    def __init__(self, profile: Profile) -> None:
        super().__init__(profile)

        # check that the storage is reachable and at the correct version
        self.migrator(profile).validate_storage()
```

i.e. you cannot initialise it without validation

Reply:
I guess, in sqlalchemy terminology, you can think of the `Profile` as the `Engine`, and the `Backend` as the `Connection`.
The engine just initialises with the connection details, but does not try to connect (so can always be initialised), but then on `Engine.connect` you try to actually initialise the `Connection`, which can only be initialised if this is possible.

Reply:
In fact, I'm tempted to rename `Backend` -> `StorageConnection` or something and, probably in https://github.com/aiidateam/aiida-core/pull/5172, add a `Profile.connect` that returns an initialised instance

Reply:
@sphuber what do you think of `StorageAbstract`, or any other suggestions? I certainly want to change it from something other than `Backend`.

As discussed, I also want to rename `aiida/backends` to `aiida/storage`, and look at moving `aiida/orm/implementation/` into here. But, in the interest of git diffs, will do that in a separate PR

Reply:
I like the ideas but I would certainly not include that here. Since that won't have consequences for end users, we can easily do this after the release, so,  as you say, let's hold off with that for now in order to not further complicate this PR